hudi-spark-datasource/hudi-spark-common/src/main/java/org/apache/hudi/HoodieSparkIndexClient.java

```diff
@@ -82,8 +82,10 @@ public static HoodieSparkIndexClient getInstance(SparkSession sparkSession) {
 
   @Override
   public void create(HoodieTableMetaClient metaClient, String indexName, String indexType, Map<String, Map<String, String>> columns, Map<String, String> options) {
-    indexName = indexType.equals(PARTITION_NAME_SECONDARY_INDEX) ? PARTITION_NAME_SECONDARY_INDEX_PREFIX + indexName : PARTITION_NAME_FUNCTIONAL_INDEX_PREFIX + indexName;
-    if (indexExists(metaClient, indexName)) {
+    String hoodieIndexName = indexType.equals(PARTITION_NAME_SECONDARY_INDEX)
```

Comment:
rename : indexName.

Reply:
I wanted to differentiate between user provided index name and internal index name. Made a change to have userIndexName and fullIndexName